rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/nodes.stats/11_indices_metrics.yml

```diff
@@ -502,7 +502,7 @@
   - gte: { nodes.$node_id.indices.mappings.total_count: 26 }
   - is_true: nodes.$node_id.indices.mappings.total_estimated_overhead
   - gte: { nodes.$node_id.indices.mappings.total_estimated_overhead_in_bytes: 26624 }
-  - match: { nodes.$node_id.indices.indices.index1.mappings.total_count: 26 }
+  - gte: { nodes.$node_id.indices.indices.index1.mappings.total_count: 26 }
```

Comment:
I relaxed the requirements of this test because it appears that the ML plugin is inconsistently loaded when these tests execute, depending on the environment.

For example, if you run this test using `./gradlew ':rest-api-spec:yamlRestTest' --tests "org.elasticsearch.test.rest.ClientYamlTestSuiteIT" -Dtests.method="test {yaml=nodes.stats/11_indices_metrics/*}"`, the ML plugin is not loaded and the mapping count is 26.

However, the CI appears to run BwC tests in such a way that the ML plugin _is_ loaded, making the mapping count 27 and causing this test to fail.

@kingherc wrote this test initially, can you check if these test changes are OK?

Reply:
Hi @Mikep86 I see. Is there a way to ensure that the ML plugin is always loaded? If not, then I think putting "gte" is fine. I would also then suggest to add the additional mapper in the comments above (in line 500) and add an explanation that the mapper appears only when the ML plugin is loaded (which in some CI pipelines it may not be).

Reply:
@kingherc It's theoretically possible to always load the ML plugin for these tests by loading it [here](https://github.com/elastic/elasticsearch/blob/f35cf032e86792fd0710f4033694e5ccb9a3820e/rest-api-spec/src/yamlRestTest/java/org/elasticsearch/test/rest/ClientYamlTestSuiteIT.java#L34), but I'm not sure we want to for a couple reasons:

- The ML plugin has a fair amount of dependencies, so it wouldn't be as simple as adding a call to load the ML plugin. We would also need to load those dependencies (and any settings required by those dependencies), which could have unintended side-effects for this test or other REST API tests.
- As far as I understand, this is intended to be more of a "core" REST API test suite, so adding a dependency on a plugin doesn't quite fit the intent. Additionally, other plugins in the future may add other metadata mappers that we would also need to load here to satisfy the test.

Given the intent of this test suite, I am of the opinion that adjusting this test to use `gte` is the best long-term fix. That way, this test checks that the expected minimum set of mappers are present, but does not fail if additional mappers from plugins are also present.

What do others think about this?

Reply:
I'd say let's use gte to check a minimum amount of field mappers. This is a test that is executed from different suites and that implies having ML plugin loaded or not.

As a middle ground, we could check that the number of mappers is both `gte` and `lte` of specific values, so we ensure that the optional metadata field mappers are taken into account, but we keep them under control as well and the test fails if we don't take into account newer created mappings.

And of course document the why in the test 😄 

WDYT?

Reply:
I like it, I will update the test accordingly